src/main.rs

```diff
@@ -214,6 +217,31 @@ where
     })
 }
 
+fn init_update_state() -> UpdateState {
+    let global_ssh_config = if let Ok(global_ssh_config_file) = File::open("/etc/ssh/ssh_config") {
+        let mut global_ssh2_config_reader = BufReader::new(global_ssh_config_file);
+        SshConfig::default()
+            .parse(
+                &mut global_ssh2_config_reader,
+                ssh2_config::ParseRule::ALLOW_UNKNOWN_FIELDS,
+            )
+            .ok()
+    } else {
+        None
+    };
+    let local_ssh_config =
+        SshConfig::parse_default_file(ssh2_config::ParseRule::ALLOW_UNKNOWN_FIELDS).ok();
+    let xdg = BaseDirectories::new().unwrap();
+    let cache_dir = xdg
+        .create_cache_directory("update-daemon")
+        .unwrap_or_else(good_panic("Failed to create a cache directory", 77));
+    return UpdateState {
+        cache_dir,
+        global_ssh_config,
+        local_ssh_config,
+    };
+}
```

Comment:
This seems a little bit cleaner than `if let` with a `None` else-branch. Also `xdg` variable is kinda redundant.

```suggestion
fn init_update_state() -> UpdateState {
    let global_ssh_config =
        File::open("/etc/ssh/ssh_config")
            .ok()
            .and_then(|global_ssh_config_file| {
                SshConfig::default()
                    .parse(
                        &mut BufReader::new(global_ssh_config_file),
                        ssh2_config::ParseRule::ALLOW_UNKNOWN_FIELDS,
                    )
                    .ok()
            });
    let local_ssh_config =
        SshConfig::parse_default_file(ssh2_config::ParseRule::ALLOW_UNKNOWN_FIELDS).ok();
    let cache_dir = BaseDirectories::new()
        .unwrap()
        .create_cache_directory("update-daemon")
        .unwrap_or_else(good_panic("Failed to create a cache directory", 77));
    UpdateState {
        cache_dir,
        global_ssh_config,
        local_ssh_config,
    }
}
```